test/benchmark/benchmark_sift.go

```diff
@@ -128,123 +129,87 @@ func readSiftFloat(file string, maxObjects int) []*models.Object {
 	return objects
 }
 
-func sendRequests(c *http.Client, request *http.Request) (*http.Response, int64, error) {
-	timeStart := time.Now()
-	response, err := c.Do(request)
-	return response, time.Since(timeStart).Milliseconds(), err
-}
-
-// If there is already a schema present, clear it out
-func clearExistingObjects(c *http.Client, url string) {
-	checkSchemaRequest, _ := http.NewRequest("GET", url+"schema", bytes.NewReader(make([]byte, 0)))
-	checkSchemaResponse, err := c.Do(checkSchemaRequest)
-	if err == nil && checkSchemaResponse.StatusCode != 200 {
-		return
-	}
-	schemaResponseBytes, _ := ioutil.ReadAll(checkSchemaResponse.Body)
-	checkSchemaResponse.Body.Close()
-
-	var dump models.Schema
-	if err := json.Unmarshal(schemaResponseBytes, &dump); err != nil {
-		panic("Could not unmarshal read response, error: " + err.Error())
-	}
-	for _, classObj := range dump.Classes {
-		requestDelete, _ := http.NewRequest("DELETE", url+"schema/"+classObj.Class, nil)
-		responseDelete, err := c.Do(requestDelete)
-		if err != nil || responseDelete.StatusCode != 200 {
-			panic("Could delete schema, error: " + err.Error())
-		}
-		responseDelete.Body.Close()
-	}
-}
-
-func benchmarkSift(c *http.Client, url string, maxObjects int) map[string]int64 {
+func benchmarkSift(c *http.Client, url string, maxObjects int) (map[string]int64, error) {
 	clearExistingObjects(c, url)
 	objects := readSiftFloat("sift_base.fvecs", maxObjects)
-	objectsJSON, _ := json.Marshal(batch{objects})
-
 	queries := readSiftFloat("sift_query.fvecs", maxObjects/100)
 	requestSchema := createSchemaSIFTRequest(url)
 
 	passedTime := make(map[string]int64)
 
 	// Add schema
-	responseSchema, timeSchema, err := sendRequests(c, requestSchema)
+	responseSchemaCode, _, timeSchema, err := performRequest(c, requestSchema)
 	passedTime["AddSchema"] = timeSchema
-	if err != nil || responseSchema.StatusCode != 200 {
-		panic("Could not add schema, error: " + err.Error())
+	if err != nil {
+		return nil, errors.New("Could not add schema, error: " + err.Error())
+	} else if responseSchemaCode != 200 {
+		return nil, errors.New("Could not add schma, http error code: " + fmt.Sprint(responseSchemaCode))
```

Comment:
here also you can change it to: `errors.Errorf("Could not add schma, http error code: %v", responseSchemaCode)`

Reply:
done!